### DISCUSSION TITLE
User & Accounts Link Schema

### DISCUSSION THREAD

**Discussion by un-focused** [nextauthjs/next-auth#1483] [Help] [answered]
I am attempting a migration from ParsePlatform to NextAuth & was looking at the PostgreSQL schema provided:

```
CREATE TABLE accounts
  (
    id                   SERIAL,
    compound_id          VARCHAR(255) NOT NULL,
    user_id              INTEGER NOT NULL,
    provider_type        VARCHAR(255) NOT NULL,
    provider_id          VARCHAR(255) NOT NULL,
    provider_account_id  VARCHAR(255) NOT NULL,
    refresh_token        TEXT,
    access_token         TEXT,
    access_token_expires TIMESTAMPTZ,
    created_at           TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
    updated_at           TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
    PRIMARY KEY (id)
  );

CREATE TABLE users
  (
    id             SERIAL,
    name           VARCHAR(255),
    email          VARCHAR(255),
    email_verified TIMESTAMPTZ,
    image          TEXT,
    created_at     TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
    updated_at     TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
    PRIMARY KEY (id)
  );
```

What I don't quite understand is why there is no foreign key constraint linking the userID in ACCOUNTS with the id in the USERS table? Are they independent from one another or should they linked? It seems they should be but I wasn't quite sure. Does anybody know the reason behind this? Or is this an issue that I should put up.

**Comment by StefanSelfTaught** [comment, marked as answer]
You can check out this 2 issues: https://github.com/nextauthjs/next-auth/issues/254 and #1235 

From #254:

> I didn't add Foreign Keys because of the potential complexity it might cause to support, particularly if it turns out it's inconsistent between databases. I was also concerned about not being able to test it throughly.
>

**Reply by un-focused** [reply]
Thanks @StefanSelfTaught! That gives more clarity. I wonder if that means they do two separate queries to link the tables (probably abstracted through the ORM library). Probably insignificant on speed as they are two simple tables.

Appreciate the information :)
